pkg/util/tar_utils.go

```diff
@@ -119,11 +122,38 @@ func unpackTar(tr *tar.Reader, path string, whitelist []string) error {
 			if err = os.Symlink(header.Linkname, target); err != nil {
 				logrus.Errorf("Failed to create symlink between %s and %s: %s", header.Linkname, target, err)
 			}
+		case tar.TypeLink:
+			linkname := filepath.Join(path, header.Linkname)
+			// Check if the linkname already exists
+			if _, err := os.Stat(linkname); !os.IsNotExist(err) {
+				// If it exists, create the hard link
+				resolveHardlink(linkname, target)
+			} else {
+				hardlinks[target] = linkname
+			}
+		}
+	}
+
+	for target, linkname := range hardlinks {
+		logrus.Info("Resolving hard links.")
+		if _, err := os.Stat(linkname); !os.IsNotExist(err) {
+			// If it exists, create the hard link
+			resolveHardlink(linkname, target)
+		} else {
+			logrus.Errorf("Unable to create hard link from %s to %s", linkname, target)
 		}
 	}
 	return nil
 }
 
+func resolveHardlink(linkname, target string) {
```

Comment:
can you return an error from this function? then you can just have the error message here and get rid of the one above. you can use `errors.Wrap(err, errMsg)`